azurerm/internal/services/apimanagement/resource_arm_api_management.go

```diff
@@ -143,6 +143,32 @@ func resourceArmApiManagementService() *schema.Resource {
 				},
 			},
 
+			"virtual_network_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(apimanagement.VirtualNetworkTypeNone),
+					string(apimanagement.VirtualNetworkTypeExternal),
+					string(apimanagement.VirtualNetworkTypeInternal),
+				}, false),
+			},
+
+			"virtual_network_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"subnet_id": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
```

Comment:
Could we validate this is a proper id with `azure.ValidateResourceID`

Reply:
Done!